src/Type/Generic/TemplateObjectType.php

```diff
@@ -0,0 +1,109 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type\Generic;
+
+use PHPStan\Type\IntersectionType;
+use PHPStan\Type\NeverType;
+use PHPStan\Type\ObjectType;
+use PHPStan\Type\Type;
+use PHPStan\Type\TypeCombinator;
+use PHPStan\Type\UnionType;
+use PHPStan\Type\VerbosityLevel;
+
+final class TemplateObjectType extends ObjectType implements TemplateType
+{
+
+	/** @var string */
+	private $name;
+
+	public function __construct(
+		string $name,
+		string $class,
+		?Type $subtractedType = null
+	)
+	{
+		parent::__construct($class, $subtractedType);
+
+		$this->name = $name;
```

Comment:
Again, source should be here.

Reply:
Also, `accepts` and `isSuperTypeOf` should be overriden like in TemplateMixedType.